sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/APISpec.groovy

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.spock
+
+import com.azure.core.test.InterceptorManager
+import com.azure.core.test.TestMode
+import com.azure.core.test.utils.TestResourceNamer
+import com.azure.core.util.configuration.ConfigurationManager
+import com.azure.core.util.logging.ClientLogger
+import com.azure.storage.queue.QueueClientBuilder
+import com.azure.storage.queue.QueueServiceClient
+import com.azure.storage.queue.QueueServiceClientBuilder
+import org.junit.Rule
+import org.junit.rules.ExpectedException
+import org.junit.rules.TestName
+import spock.lang.Specification
+
+class APISpec extends Specification {
+    // Field common used for all APIs.
+    def logger = new ClientLogger(APISpec.class)
+    def AZURE_TEST_MODE = "AZURE_TEST_MODE"
+    def interceptorManager
+    def testResourceName
+
+    // Clients for API tests
+    def primaryQueueServiceClient
+    def primaryQueueServiceAsyncClient
+    def primaryQueueClient
+    def primaryQueueAsyncClient
+
+
+    // Test name for test method name.
+    @Rule
+    TestName testName = new TestName()
```

Comment:
Are there any collisions on test name?

Reply:
I tried the best to name it different. There is no conflicts currently